src/improvements/ci.just

```diff
@@ -0,0 +1,114 @@
+# Justfile commands used in CI
```

Comment:
Basically most of this file is simply moved from `justfile`. The idea was to reduce the size of justfile and keep only user invoked commands there. Anything ci relies on we can put in `ci.just`.